contracts/0.8.25/vaults/VaultFactory.sol

```diff
@@ -0,0 +1,109 @@
+// SPDX-FileCopyrightText: 2024 Lido <info@lido.fi>
+// SPDX-License-Identifier: GPL-3.0
+
+import {UpgradeableBeacon} from "@openzeppelin/contracts-v5.0.2/proxy/beacon/UpgradeableBeacon.sol";
+import {BeaconProxy} from "@openzeppelin/contracts-v5.0.2/proxy/beacon/BeaconProxy.sol";
+import {Clones} from "@openzeppelin/contracts-v5.0.2/proxy/Clones.sol";
+
+import {IStakingVault} from "./interfaces/IStakingVault.sol";
+
+pragma solidity 0.8.25;
+
+interface IDelegation {
+    struct InitializationParams {
+        uint256 managementFee;
+        uint256 performanceFee;
+        address manager;
+        address operator;
+    }
+
+    function DEFAULT_ADMIN_ROLE() external view returns (bytes32);
+
+    function MANAGER_ROLE() external view returns (bytes32);
+
+    function OPERATOR_ROLE() external view returns (bytes32);
+
+    function LIDO_DAO_ROLE() external view returns (bytes32);
+
+    function initialize(address admin, address stakingVault) external;
+
+    function setManagementFee(uint256 _newManagementFee) external;
+
+    function setPerformanceFee(uint256 _newPerformanceFee) external;
+
+    function grantRole(bytes32 role, address account) external;
+
+    function revokeRole(bytes32 role, address account) external;
+}
+
+contract VaultFactory is UpgradeableBeacon {
+    address public immutable delegationImpl;
```

Comment:
```suggestion
    address public immutable DELEGATION_IMPL;
```

Reply:
Why it's immutable?

Reply:
should think about disentangling the beacon and factory (from Azat)

Reply:
@loga4 (the problem is that it's impossible to upgrade the delegation for the factory)